diff --git a/dist/index.js b/dist/index.js
index ea8deaafe77d6da0cf14a59d2d027f5df8633387..4013b60c06d3aae94c9295c618e3cc18a64bc96e 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -65,6 +65,7 @@ var useScrollPosition = (props) => {
       target.removeEventListener("scroll", handleScroll);
       if (throttleTimeout.current) {
         clearTimeout(throttleTimeout.current);
+        throttleTimeout.current = null;
       }
     };
   }, [elementRef == null ? void 0 : elementRef.current, delay, handler, isEnabled]);
diff --git a/dist/index.mjs b/dist/index.mjs
index 9c46bad08dafb3cbdcd2be7b03a828fe803d4833..795e7997c014fa13b15fe3a4951e1da9cf4e427e 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -41,6 +41,7 @@ var useScrollPosition = (props) => {
       target.removeEventListener("scroll", handleScroll);
       if (throttleTimeout.current) {
         clearTimeout(throttleTimeout.current);
+        throttleTimeout.current = null;
       }
     };
   }, [elementRef == null ? void 0 : elementRef.current, delay, handler, isEnabled]);
